Sources_v3/Query/ChannelQuery.swift

```diff
@@ -61,7 +61,19 @@ public struct ChannelQuery<ExtraData: ExtraDataTypes>: Encodable {
         watchersPagination = []
         self.options = options
     }
-    
+
+    /// Init a channel query.
+    /// - Parameters:
+    ///   - cid: New `ChannelId` for channel query..
+    ///   - channelQuery: ChannelQuery with old cid.
+    init(cid: ChannelId, channelQuery: Self) {
+        self.init(cid: cid,
+                  messagesPagination: channelQuery.messagesPagination,
+                  membersPagination: channelQuery.membersPagination,
+                  watchersPagination: channelQuery.watchersPagination,
+                  options: channelQuery.options)
+    }
+
```

Comment:
I wanted to suggest making `cid` mutable but then we could end up with inconsistency between `cid` and `channelPayload `. So I guess this is the best option 👍

Reply:
I was considering this option, but I also don't like idea of mutable `id` in object in general.